src/NLPModelsIpopt.jl

```diff
@@ -181,6 +182,53 @@ function ipopt(nlp::AbstractNLPModel; kwargs...)
   return solve!(solver, nlp, stats; kwargs...)
 end
 
+"""
+    ipopt(nls::AbstractNLSModel; kwargs...)
+
+Solve the least-squares problem `nls` using `IPOPT` by moving the nonlinear residual to the constraints.
+
+# Arguments
+- `nls::AbstractNLSModel`: The least-squares problem to solve.
+
+For advanced usage, first define an `IpoptSolver` to preallocate the memory used in the algorithm, and then call `solve!`:
+    solver = IpoptSolver(nls)
+    solve!(solver, nls; kwargs...)
+
+# Examples
+```julia
+using NLPModelsIpopt, ADNLPModels
+nls = ADNLSModel(x -> [x[1] - 1, x[2] - 2], [0.0, 0.0], 2)
+stats = ipopt(nls, print_level = 0)
+```
+"""
+function ipopt(ff_nls::FeasibilityFormNLS; kwargs...)
+  solver = IpoptSolver(ff_nls)
+  stats = GenericExecutionStats(ff_nls)
+  stats = solve!(solver, ff_nls, stats; kwargs...)
+
+  return stats
+end
+
+function ipopt(nls::AbstractNLSModel; kwargs...)
+  ff_nls = FeasibilityFormNLS(nls)
+  stats = ipopt(ff_nls; kwargs...)
+
+  stats.solution = stats.solution[1:nls.meta.nvar]
+
+  if hasproperty(stats, :multipliers_L)
```

Comment:
No need to check if the stats has the fields.
However, you can update this if `has_bounds(nls)` and update the `stats.multipliers` if nls.meta.ncon >0